src/routes/MarketDetails/store/actions/buyShares.js

```diff
@@ -60,41 +62,39 @@ const buyMarketShares = (market, outcomeIndex, outcomeTokenCount, cost) => async
   }
   if (approvalResetAmount) transactions.unshift(SETTING_ALLOWANCE)
 
-  const payload = await fetchMarket(market.address)
-  const updatedMarket = payload.entities.markets[market.address]
+  const updatedMarket = await requestFromRestAPI(`markets/${market.address}`)
   const updatedPrice = updatedMarket.marginalPrices[outcomeIndex]
-  const oldPrice = market.marginalPrices[outcomeIndex]
+  const oldPrice = market.outcomes.toArray()[outcomeIndex].marginalPrice
   if (!allowedRangePrice(oldPrice, updatedPrice)) {
     dispatch(openModal({ modalName: 'ModalOutcomePriceChanged' }))
-    return dispatch(receiveEntities(payload))
+    return processMarketResponse(dispatch, getState(), updatedMarket)
```

Comment:
@mmvsha Here we should dispatch an action specifically to update the price inside this market, what do you think?

Reply:
@andre-meyer it’s dispatched inside `processMarketResponse` 🙂